File: C++/206. Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* temp=head;
        ListNode* pre=NULL;
        ListNode* a;
        while(temp){
            a=temp->next;
            temp->next=pre;
            pre=temp;
            temp=a;
        }
        return pre;
    }
};

File: JAVA/206. Reverse Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode temp=head;
        ListNode pre=null;
        ListNode a;
        while(temp!=null){
            a=temp.next;
            temp.next=pre;
            pre=temp;
            temp=a;
        }
        return pre;
    }
}